tmt/convert.py

```diff
@@ -109,47 +110,80 @@ def read(path, makefile, nitrate, purpose):
         if not testcases:
             raise ConvertError("No testcase found for '{0}'.".format(test))
         elif len(testcases) > 1:
-            log.warn("Multiple test cases found for '{0}', using {1}".format(
-                test, testcases[0].identifier))
-        testcase = testcases[0]
-        echo("test case found '{0}'.".format(testcase.identifier))
-        # Contact
-        if testcase.tester:
-            data['contact'] = '{} <{}>'.format(
-                testcase.tester.name, testcase.tester.email)
-            echo(style('contact: ', fg='green') + data['contact'])
-        # Environment
-        if testcase.arguments:
-            data['environment'] = tmt.utils.variables_to_dictionary(
-                testcase.arguments)
-            echo(style('environment:', fg='green'))
-            echo(pprint.pformat(data['environment']))
-        # Relevancy
-        field = tmt.utils.StructuredField(testcase.notes)
-        data['relevancy'] = field.get('relevancy')
-        echo(style('relevancy:', fg='green'))
-        echo(data['relevancy'].rstrip('\n'))
+            log.warn("Multiple test cases found for '{0}'.".format(
+                test))
+
+        testcase_data = list()
+
+        for testcase in testcases:
+            single_case_data = dict()
```

Comment:
Seems the `read()` function is getting a little bit too large. Would it make sense to create a separate function `read_tmcs()` and call it from here?

Reply:
In theory yes, however I don't really see someone using it other than `read()` as it is pretty dependant on it.